### PR TITLE
TPC-H datagenerator and instructions

### PR DESCRIPTION
**Pull request by npoggi**
Adds full TPCH support in spark-sql-perf
- Basic TPCH partitioning following the VectorH paper as a baseline.
- New multi-benchmark and multi-scale factor for TPCH and TPCDS.
- TPCH running script
- Updates to the README for TPCH

### CODE REVIEW

File: src/main/notebooks/TPC-multi_datagen.scala [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+    println("Elapsed time: " + (t1 - t0) + "ms")
+    result
+}
+
+// COMMAND ----------
+
+// FOR INSTALLING TPCH DBGEN (with the sdtout patch)
+def installDBGEN(url: String = "https://github.com/databricks/tpch-dbgen.git", useStdout: Boolean = true, baseFolder: String = "/tmp")(i: java.lang.Long): String = {
+  // check if we want the revision which makes dbgen output to stdout
+  val checkoutRevision: String = if (useStdout) "git checkout 0469309147b42abac8857fa61b4cf69a6d3128a8 -- bm_utils.c" else ""
```
Comment (juliuszsompolski) [inline src/main/notebooks/TPC-multi_datagen.scala]: would we want not to do that?
we always do this tweak with stdout for dsdgen, so maybe this option here adds complexity, and adds a well buried dependency on a concrete githash in the repo.

File: src/main/notebooks/TPC-multi_datagen.scala [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+  // Avoid OOM when shuffling large scale fators
+  // and errors like 2GB shuffle limit at 10TB like: Most recent failure reason: org.apache.spark.shuffle.FetchFailedException: Too large frame: 9640891355
+  // For 10TB 16x4core nodes were needed with the config below, 8x for 1TB and below. 
+  // About 24hrs. for SF 1 to 10,000.
+  if (scaleFactor.toInt >= 10000) {    
+    spark.conf.set("spark.sql.shuffle.partitions", "20000")
+    SparkHadoopUtil.get.conf.set("parquet.memory.pool.ratio", "0.1")
+  } 
+  else if (scaleFactor.toInt >= 1000) {
+    spark.conf.set("spark.sql.shuffle.partitions", "2001") //one above 2000
```
Comment (juliuszsompolski) [inline src/main/notebooks/TPC-multi_datagen.scala]: why? :)

File: src/main/notebooks/TPC-multi_datagen.scala [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+  }
+  else { 
+    spark.conf.set("spark.sql.shuffle.partitions", "200") //default
+    SparkHadoopUtil.get.conf.set("parquet.memory.pool.ratio", "0.5")
+  }
+}
+
+// COMMAND ----------
+
+// Gernerate the data, import the tables, generate stats for selected benchmarks and scale factors
```
Comment (juliuszsompolski) [inline src/main/notebooks/TPC-multi_datagen.scala]: Gernerate -> Generate

File: src/main/notebooks/tpch_run.scala [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  s"s3a://my-bucket/tpch/sf${scaleFactor}_${format}"
+
+val resultLocation = "s3a://my-bucket/results"
+val iterations = 2
+def databaseName(scaleFactor: Int, format: String) = s"tpch_sf${scaleFactor}_${format}"
+val randomizeQueries = false //to use on concurrency tests
+
+// Experiment metadata for results, edit if outside Databricks
+val configuration = "default" //use default when using the out-of-box config
+val runtype = "Azure Databricks TPCH" + spark.conf.get("spark.databricks.clusterUsageTags.sparkVersion")
```
Comment (juliuszsompolski) [inline src/main/notebooks/tpch_run.scala]: leave out `Azure Databricks` for this example notebook?

File: src/main/notebooks/TPC-multi_datagen.scala (hunk 1 above)
Comment (npoggi) [reply src/main/notebooks/TPC-multi_datagen.scala]: We might want to keep the source csv files or test the full import from file system. I prefer to have this option available.

File: src/main/notebooks/TPC-multi_datagen.scala (hunk 2 above)
Comment (npoggi) [reply src/main/notebooks/TPC-multi_datagen.scala]: To use `HighlyCompressedMapStatus` https://github.com/apache/spark/blob/4be33758354e1f95fd1d82a5482f3f00218e8c91/core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala#L52-L58

File: src/main/notebooks/tpch_run.scala (hunk 4 above)
Comment (npoggi) [reply src/main/notebooks/tpch_run.scala]: Removed

**Pull request merged**